### ISSUE TITLE
make mapM_ concurrent

### ISSUE THREAD

**Issue by harendra-kumar** [composewell/streamly#82]
`mapM_` can be made concurrent just like `mapM` is.

**Comment by harendra-kumar** [comment]
We can write a concurrent version just as : `runStream . mapM` and the type will change from:

```
mapM_ :: Monad m => (a -> m b) -> SerialT m a -> m ()
```
To

```
mapM_ :: (IsStream t, Monad m) => (a -> m b) -> t m a -> m ()
```

But with this we will be back to the problem of ambiguous types discussed in #33 and fixed via #61. 
We need to rethink that problem and how to resolve this in general. Earlier we assumed that fold functions are serial, but now we are going make many of those concurrent.

**Comment by harendra-kumar** [comment]
This is irrelevant with the new concurrency design in 0.9 . We have only a few fundamental concurrency primitives.
